pkg/diagnose/diagnosis/loader.go

```diff
@@ -0,0 +1,27 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+package diagnosis
+
+import log "github.com/cihub/seelog"
+
+// Catalog holds available diagnosis for detection and usage
+type Catalog map[string]Diagnosis
+
+// DefaultCatalog holds every compiled-in diagnosis
+var DefaultCatalog = make(Catalog)
+
+// Register a diagnosis that will be called on diagnose
+func Register(name string, d Diagnosis) {
+	if _, ok := DefaultCatalog[name]; ok {
+		log.Warnf("Diagnosis %s already registered, overriding it", name)
+	}
+	DefaultCatalog[name] = d
+}
+
+// Diagnosis should implement Diagnose to report status
+type Diagnosis interface {
```

Comment:
I would have gone with 
`type Diagnosis func() error`, and only registering callbacks, to make it easier to implement. That way, packages can register bare functions without creating a struct.

Having a struct is good for keeping a state, but IIRC we want these diags to be completely stateless. Am I wrong?

Reply:
Good idea, I'll change this 👍